unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTAsset/NFTAssetLoadHelper.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Helpers;
+using NFTShape_Internal;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace DCL
+{
+    public interface INFTAssetLoadHelper : IDisposable
+    {
+        IEnumerator LoadNFTAsset(string url, Action<INFTAsset> OnSuccess, Action<Exception> OnFail);
+    }
+
+    public class NFTAssetLoadHelper : INFTAssetLoadHelper
+    {
+        protected AssetPromise_Texture imagePromise = null;
+        protected AssetPromise_Gif gifPromise = null;
+
+        public IEnumerator LoadNFTAsset(string url, Action<INFTAsset> OnSuccess, Action<Exception> OnFail)
+        {
+            if (string.IsNullOrEmpty(url))
+            {
+                OnFail?.Invoke(new Exception($"Image url is null!"));
+                yield break;
+            }
+
+            const string CONTENT_TYPE = "Content-Type";
+            const string CONTENT_LENGTH = "Content-Length";
+            const string CONTENT_TYPE_GIF = "image/gif";
+            const long PREVIEW_IMAGE_SIZE_LIMIT = 500000;
+
+            HashSet<string> headers = new HashSet<string>() {CONTENT_TYPE, CONTENT_LENGTH};
+            Dictionary<string, string> responseHeaders = new Dictionary<string, string>();
+            string headerRequestError = string.Empty;
+
+            yield return GetHeaders(url, headers, result => responseHeaders = result, (x) => headerRequestError = x);
+
+            if (!string.IsNullOrEmpty(headerRequestError))
+            {
+                OnFail?.Invoke(new Exception($"Error fetching headers! ({headerRequestError})"));
+                yield break;
+            }
+
+            Debug.Log("Loading preview... " + url);
+            string contentType = responseHeaders[CONTENT_TYPE];
+            long contentLength = long.Parse(responseHeaders[CONTENT_LENGTH]);
```

Comment:
could we use `TryParse` here? an issue on the server side might result on the response header not containing `Content-Length` it's not very common but it might be better to prevent exception here